cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/SparkProgramController.java

```diff
@@ -35,10 +37,8 @@
     this.context = context;
   }
 
-  /**
-   * Returns the {@link SparkContext} for Spark run represented by this controller.
-   */
-  public SparkContext getContext() {
-    return context;
+  @Override
+  public WorkflowToken getWorkflowToken() {
+    return context.getWorkflowToken();
```

Comment:
I think checks that are performed in the `MapReduceProgramController` should also be here.